<?xml version="1.0" encoding="utf-8"?>
<launch>

  <arg name="debug" default="false"/>
  <arg if="$(arg debug)" name="launch_prefix" value="xterm -rv -e gdb -ex run -args"/>
  <arg if="$(arg debug)" name="respawn" value="false"/>
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>
  <arg unless="$(arg debug)" name="respawn" value="true"/>

  <arg name="configfile" default="./config_euroc_segmentation.yaml" />
 
  <arg name="cloud_subsample_factor" default="1" />
  <arg name="process_frame" default="/camera_rgb_optical_frame"/>

<node pkg="spacebot_kinect" type="segmentation_node" name="segmentation_node" respawn="$(arg respawn)" launch-prefix="$(arg launch_prefix)">
    <rosparam command="load" file="$(arg configfile)" />   
    <param name="input_cloud"    value="/camera/depth_registered/points" />
    <param name="cloud_subsample_factor"	value="$(arg cloud_subsample_factor)" />
    <param name="debug_output"   value="true" />
    <param name="start_paused"   value="true" />
    <param name="process_frame"   value="$(arg process_frame)" />
  </node>
  
  <node pkg="rosservice" type="rosservice" name="start_screw_segm_service" args="call --wait /segmentation_node/service_startstop 'start: 1'"/>
  
  
  
</launch>
